doc/build-helpers/testers.chapter.md

```diff
@@ -169,15 +169,18 @@ The build will fail if `shellcheck` finds any issues.
 
 Checks that the output from running a command contains the specified version string in it as a whole word.
 
-NOTE: In most cases, [`versionCheckHook`](#versioncheckhook) should be preferred, but this function is provided and documented here anyway. The motivation for adding either tests would be:
+Although being very simple and naive, this test is useful:
 
-- Catch dynamic linking errors and such and missing environment variables that should be added by wrapping.
-- Probable protection against accidentally building the wrong version, for example when using an "old" hash in a fixed-output derivation.
+- It assures the main program can run.
+- It catches dynamic linking errors, missing or malformed environment variables and other runtime errors.
```

Comment:
You know the word "catches" doesn't fit a `passthru.tests.version` check, because if you won't build that `passthru.tests` derivation you won't catch anything. I'd use the word "catches" when describing `versionCheckHook`, but not when describing both.

Reply:
[You approved that wording before](https://github.com/NixOS/nixpkgs/pull/348025#discussion_r1807456082).

Reply:
> [You approved that wording before](https://github.com/NixOS/nixpkgs/pull/348025#discussion_r1807456082).

Correct, I inspected your phrasing in more details and changed my mind.

> Further, I don't think this difference matters here.
> 
> `passthru.tests.version` catches errors when executed. `versionCheckHook` catches errors when executed.
> 
> The difference is `passthru.tests.version` is executed directly when called e.g. via commandline, while `versionCheckHook` is executed indirectly when the package containing it is built.

That's a big difference IMO. Of course we agree that if you'd explicitly build `passthru.tests.version` you'd catch an issue with the package, but I find it inaccurate to suggest that merely adding a `passthru.tests.version` will make any such issue be catched, because all the reasons you are already aware of from [our discussion on Discourse](https://discourse.nixos.org/t/improve-the-jdk-infrastructure-on-nixpkgs/45678/21).

Reply:
Please suggest a better phrasing, since this word was present on the document already.

Reply:
> Please suggest a better phrasing, since this word was present on the document already.

Just to be clear, I wouldn't have said anything if these words weren't present in the diff.. But I'd use the words:

> It can bring to awareness dynamic linking errors, missing or malformed environment variables and other runtime errors.

Reply:
Done.